eth/backend.go

```diff
@@ -544,61 +544,127 @@ func (s *Ethereum) waitForSyncAndMaxwell(parlia *parlia.Parlia) {
 				continue
 			}
 			log.Info("Node is synced and Maxwell fork is active, proceeding with node ID registration")
-			err := s.registerNodeID(parlia)
+			err := s.updateNodeID(parlia)
 			if err == nil {
 				return
 			}
 			retryCount++
 			if retryCount > 3 {
-				log.Error("Failed to register node ID exceed max retry count", "retryCount", retryCount, "err", err)
+				log.Error("Failed to update node ID exceed max retry count", "retryCount", retryCount, "err", err)
 				return
 			}
 		}
 	}
 }
 
-// registerNodeID registers the node ID with the StakeHub contract
-func (s *Ethereum) registerNodeID(parlia *parlia.Parlia) error {
-	// Check if node ID is already registered
-	nodeIDs, err := parlia.GetNodeIDs()
+// updateNodeID registers the node ID with the StakeHub contract
+func (s *Ethereum) updateNodeID(parlia *parlia.Parlia) error {
+	nonce, err := s.APIBackend.GetPoolNonce(context.Background(), s.etherbase)
 	if err != nil {
-		log.Error("Failed to get registered node IDs", "err", err)
+		return fmt.Errorf("failed to get nonce: %v", err)
+	}
+
+	// Handle removals first
+	if err := s.handleRemovals(parlia, nonce); err != nil {
 		return err
 	}
+	nonce++
+
+	// Handle additions
+	return s.handleAdditions(parlia, nonce)
+}
+
+func (s *Ethereum) handleRemovals(parlia *parlia.Parlia, nonce uint64) error {
+	if len(s.config.EVNNodeIDsToRemove) == 0 {
+		return nil
+	}
 
-	// Check which node IDs need to be registered
-	nodeIDsToAdd := make([]enode.ID, 0)
-	for _, idToRegister := range s.config.ValidatorNodeIDsToAdd {
-		isRegistered := false
-		for _, id := range nodeIDs {
-			if id == idToRegister {
-				isRegistered = true
-				break
+	// Handle wildcard removal
+	if len(s.config.EVNNodeIDsToRemove) == 1 {
```

Comment:
Does removal need to check the `registeredSet`? It may avoid a repeat removal operation.

Reply:
After the nodeIDs have been removed, they may be removed again when the BSC is restarted next time. If here check the `registeredSet` from on-chain, it may avoid this repeat removal operation.

Reply:
It's not a bug, if you agree, you could optimize it next time.